operator/src/test/java/org/keycloak/operator/testsuite/integration/CacheTest.java

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.operator.testsuite.integration;
+
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.quarkus.logging.Log;
+import io.quarkus.test.junit.QuarkusTest;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.crds.v2alpha1.deployment.Keycloak;
+import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakStatusCondition;
+import org.keycloak.operator.crds.v2alpha1.deployment.spec.CacheSpecBuilder;
+import org.keycloak.operator.testsuite.utils.CRAssert;
+import org.keycloak.operator.testsuite.utils.K8sUtils;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.testsuite.utils.K8sUtils.deployKeycloak;
+
+@QuarkusTest
+public class CacheTest extends BaseOperatorTest {
+
+    private static final String CONFIGMAP_NAME = "my-config";
+
+    @AfterEach
+    public void cleanupConfigMap() {
+        k8sclient.configMaps().withName(CONFIGMAP_NAME).delete();
+    }
+
+    @Test
+    public void testCreateCacheConfigMapFileAfterDeployment() {
+        var kc = getTestKeycloakDeployment(true);
+        var deploymentName = kc.getMetadata().getName();
+        kc.getSpec().setCacheSpec(new CacheSpecBuilder().withNewConfigMapFile("file", CONFIGMAP_NAME, false).build());
+
+        deployKeycloak(k8sclient, kc, false);
+
+        // Check Operator has deployed Keycloak and the statefulset exists, this allows
+        // for the watched configmap to be picked up
+        Log.info("Checking Operator has deployed Keycloak deployment");
+        Resource<StatefulSet> stsResource = k8sclient.resources(StatefulSet.class).withName(deploymentName);
+        Resource<Keycloak> keycloakResource = k8sclient.resources(Keycloak.class).withName(deploymentName);
+        // expect no errors and not ready, which means we'll keep reconciling
+        Awaitility.await().ignoreExceptions().atMost(2, TimeUnit.MINUTES).untilAsserted(() -> {
+            assertThat(stsResource.get()).isNotNull();
+            Keycloak keycloak = keycloakResource.get();
+            CRAssert.assertKeycloakStatusCondition(keycloak, KeycloakStatusCondition.HAS_ERRORS, false);
+            CRAssert.assertKeycloakStatusCondition(keycloak, KeycloakStatusCondition.READY, false);
+        });
+
+        createCacheConfigMap();
+
+        K8sUtils.waitForKeycloakToBeReady(k8sclient, kc);
+    }
+
+    @Test
+    public void testCacheConfigMapFile() {
+        var kc = getTestKeycloakDeployment(true);
```

Comment:
Would it make sense to run with probes enabled in this test?
```suggestion
        var kc = getTestKeycloakDeployment(false);
```

Reply:
Not necessarily - the tests here are really just to exercise the operator's handling of configmaps. We could add a polling test as well here. All of the other considerations - how the file is mounted, if it's valid - are either covered by other tests or aren't really the concern of this feature.

Reply:
Makes sense, thanks. Let's not hold this PR any longer. LGTM.